openshift/machine-config-operator
# MCO-746: make buildcontroller tests less flaky

**- What I did**

This makes the BuildController unit tests less flaky in a couple of ways:
1. Consolidates all MachineConfigPool object interrogation / mutation into a single place.
2. Ensures that MachineConfigPool object updates are retried in the event of a conflict. This work was imported from https://github.com/openshift/machine-config-operator/pull/3894 and should be merged as part of this PR.

**- How to verify it**

To verify this, the unit test suite must be run exhaustively. To reproduce this locally, one can clone this PR and run: `$ go test -count=100 -failfast ./pkg/controller/build` This will effectively run the unit test suite 100 times. While I cannot completely guarantee that it will never fail, the likelihood that it will is substantially reduced.

**- Description for the changelog**
Stabilizes BuildController unit test suite
